contracts/Commitment/Commitment.sol

```diff
@@ -210,22 +262,57 @@ contract Commitment is
         payable
         withTimedTransitions()
         onlyStates(State.Whitelist, State.Public)
+
+        // AUDIT[CHF-55] Accepting agreement confusion.
+        //   This will automatically put the msg.sender in the list of addresses
+        //   accepting the agreement. Shouldn't the order of actions be
+        //   different? Accepting the agreement being a precondition of commit()
+        //   action?
         acceptAgreement(msg.sender)
     {
         // Take with EtherToken allowance (if any)
         uint256 commitedWei = ETHER_TOKEN.allowance(msg.sender, this);
+
+        // AUDIT[CHF-54] Unnecessary call to ETHER_TOKEN.transferFrom().
+        //   When commitedWei is 0, calling ETHER_TOKEN.transferFrom() will
+        //   - waste some amount of gas,
+        //   - produce Transfer event with value 0.
+        //   Checking the value of commitedWei before calling transferFrom()
+        //   seems reasonable.
         assert(ETHER_TOKEN.transferFrom(msg.sender, this, commitedWei));
 
         // Turn msg.value into EtherToken (if any)
+        // AUDIT[CHF-56] Unnecessary call to ETHER_TOKEN.deposit().
+        //   Similar to AUDIT[CHF-54]. When msg.value is 0, calling
+        //   ETHER_TOKEN.deposit() will
+        //   - waste some amount of gas,
+        //   - produce LogDeposit event with value 0.
+        //   - produce Transfer event from address 0 with value 0.
+        // Wrap the following code with `if (msg.value > 0)` condition.
         commitedWei = add(commitedWei, msg.value);
         ETHER_TOKEN.deposit.value(msg.value)();
 
         // Get Neumark reward
         uint256 commitedEur = convertToEur(commitedWei);
+        // AUDIT[CHF-57] Comment about 0-value commitment.
+        //   Add a comment that commitToken() will fail if commitedEur is 0
+        //   or smaller than MIN_TICKET_EUR.
         var (investorNmk, ticketNmk) = commitToken(commitedEur, Token.Ether);
+        // AUDIT[CHF-58] Move NMK counters updates to Commitment.commitToken().
+        //   The Commitment.commitToken() has all the logic related to whitelist
+        //   checking. Move the _whitelistEtherNmk subtraction from here and
+        //   _whitelistEuroNmk subtraction from commitEuro() there to
+        //   commitToken() too. This will also simplify the return type of
+        //   commitToken().
         _whitelistEtherNmk = sub(_whitelistEtherNmk, ticketNmk);
 
         // Lock EtherToken
+        // AUDIT[CHF-60] Consider using ERC223-like transfer pattern.
```

Comment:
We had long debate over it here. Personally I do not like ERC223 because you lose explicit semantics in the contract receiving tokens. It's ok to use it on simple token transfers where `transfer` is the main operation of the contract. Here the function is to lock funds. `ETHER_TOKEN.transfer(...)` does not say anything what is being done in the lock. We'd need to encode `investorNmk` in the data field as well.